test_conformance/math_brute_force/unary.cpp

```diff
@@ -491,13 +491,13 @@ static cl_int TestFloat( cl_uint job_id, cl_uint thread_id, void *data )
         if ( strcmp(fname,"sin")==0 || strcmp(fname,"cos")==0 )  //the domain of the function is [-pi,pi]
         {
           if( fabs(p_j) > M_PI )
-            p[j] = NAN;
+            ((float*)p)[j] = M_PI;
         }
 
         if ( strcmp( fname, "reciprocal" ) == 0 )
         {
           if( fabs(p_j) > 0x7E800000 ) //the domain of the function is [2^-126,2^126]
-            p[j] = NAN;
+            ((float*)p)[j] = 0x7E800000;
```

Comment:
Presumably this results in an integer to floating-point conversion which is probably not what you want.

Reply:
Thanks, I'll check the integer to floating point comparison. 

As to the verification approach being proposed, if the generated input was outside the domain of acceptable inputs, we are setting it to one that is within (in this case the last in the range of acceptable inputs). This should void the need for any special case handling in the verification code as well.  Does that address the concerns you had on the previous pull request?

Reply:
Addressed the comment about the cast